src/modules/commander/Commander.cpp

```diff
@@ -1070,13 +975,60 @@ Commander::handle_command(vehicle_status_s *status_local, const vehicle_command_
 		cmd_result = handle_command_motor_test(cmd);
 		break;
 
+	case vehicle_command_s::VEHICLE_CMD_PREFLIGHT_REBOOT_SHUTDOWN:
+
+		// do nothing for autopilot
+		if (((int)(cmd.param1)) == 0) {
+			answer_command(cmd, vehicle_command_s::VEHICLE_CMD_RESULT_ACCEPTED, command_ack_pub);
+
+			break;
+		}
+
+		if (shutdown_if_allowed()) {
```

Comment:
Note: using `shutdown_if_allowed()` here changes the behavior: Shutting down via command ...
- ...when killed but still armed not possible anymore.
- ...when the safety switch is pressed while armed is not possible anymore.

This makes total sense and is more consistent with low battery shutdown from my perspective. If these workaround cases needed to be there the arming state machine would be broken. I still wanted to point it out. 👍